### PR TITLE
[fabric]  Add orderer/peer update

### PR DESCRIPTION
**Pull request by mgCepeda**
Signed-off-by: mgCepeda <marina.gomez.cepeda@accenture.com>

**Changelog**

- Update platforms/hyperledger-fabric/configuration/add-orderer.yaml and platforms/hyperledger-fabric/configuration/add-peer.yaml: 

 

**Reviewed by**
@suvajit-sarkar
@jagpreetsinghsasan

### CODE REVIEW

**Comment by jagpreetsinghsasan** [comment]
@mgCepeda Please add the issue number in this PR description.

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml [hunk 1]
```diff
@@ -115,25 +115,81 @@
   when: network.env.proxy == 'ambassador'
 
 ############################################################################################
-# This task copies the msp folder from the respective CA Tools CLI
-# to the Ansible container
-- name: Copy the msp folder from the ca tools
+# This task copies the msp admincerts from vault
+- name: Copy the msp admincerts from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml:119]: Please rename the task and the above comment to `Fetch the msp admincerts from vault`

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml [hunk 2]
```diff
@@ -115,25 +115,81 @@
...
+    vault kv get -field=admincerts {{ vault.secret_path | default('secretsv2') }}/crypto/ordererOrganizations/{{ component_name }}/users/admin/msp > Admin@{{ component_name }}-cert.pem
+    mkdir -p ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/admincerts/
+    mv Admin@{{ component_name }}-cert.pem ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/admincerts/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  
+############################################################################################
+# This task copies the msp cacerts from vault
+- name: Copy the msp cacerts from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml:130]: Please rename the task and the above comment to `Fetch the msp cacerts from vault`

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml [hunk 3]
```diff
@@ -115,25 +115,81 @@
...
+    mkdir -p ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/cacerts/
+    mv ca-{{ component_name }}-{{ item.external_url_suffix }}-8443.pem ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/cacerts/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy != 'none'  
+
+############################################################################################
+# This task copies the msp tlscacerts from vault
+- name: Copy the msp tlscacerts from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml:142]: Please rename the task and the above comment to `Fetch the msp tlscacerts from vault`

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml [hunk 4]
```diff
@@ -115,25 +115,81 @@
...
+    mkdir -p ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/tlscacerts/
+    mv ca-{{ component_name }}-{{ item.external_url_suffix }}-8443.pem ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/tlscacerts/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy != 'none'
+
+############################################################################################
+# This task copies the msp cacerts from vault proxy is none
+- name: Copy the msp cacerts from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml:154]: Please rename the task and the above comment to `Fetch the msp cacerts from vault`

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml [hunk 5]
```diff
@@ -115,25 +115,81 @@
...
+    mkdir -p ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/cacerts/
+    mv ca-{{ component_name }}-{{ item.services.ca.grpc.port }}.pem ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/cacerts/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy == 'none'  
+
+############################################################################################
+# This task copies the msp tlscacerts from vault when proxy is none
+- name: Copy the msp tlscacerts from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml:166]: Please rename the task and the above comment to `Fetch the msp tlscacerts from vault`

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml [hunk 6]
```diff
@@ -115,25 +115,81 @@
...
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy == 'none'     
 
 ############################################################################################
-# This task copies the tls server.crt file from the respective CA Tools CLI 
+# This task copies the tls server.crt from vault
 # to the build directory
-- name: Copy the tls server.crt file from the ca tools
+- name: Copy the tls server.crt from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml:179]: Please rename the task and the above comment to `Fetch the tls server.crt from vault`

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml [hunk 7]
```diff
@@ -146,10 +202,13 @@
 ############################################################################################
 # This task copies the tls ca.crt file from the respective CA Tools CLI
 # to the address specified in network.yaml
-- name: Copy the tls ca.crt file from the ca tools
+- name: Copy the tls ca.crt file from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml:205]: Please rename the task and the above comment to `Fetch the tls ca.crt from vault`

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/peer/tasks/main.yaml [hunk 8]
```diff
@@ -182,12 +205,63 @@
     loop_var: channel
 
 ############################################################################################
-# This task copies the msp folder from the respective CA Tools CLI
-# to the build directory
-- name: Copy the msp folder from the ca tools
+# This task copies the msp admincerts from vault
+- name: Copy the msp admincerts from vault
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/peer/tasks/main.yaml:209]: Make all such statements as "Fetch from vault"

File: platforms/hyperledger-fabric/configuration/roles/create/users/tasks/main.yaml [hunk 9]
```diff
@@ -32,17 +32,25 @@
   loop_control:
     loop_var: files
 
+- pause:
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/users/tasks/main.yaml:35]: Do we need this?

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Comment by jagpreetsinghsasan** [comment]
@mgCepeda can you squash merge the commits?

**Comment by jagpreetsinghsasan** [comment]
Rest everything looks good to me.

**Review by suvajit-sarkar — APPROVED** [review]

**Pull request merged**